infra/gcp/terraform/k8s-infra-prow-build/serviceaccounts.tf

```diff
@@ -26,16 +26,38 @@ locals {
       description       = "sync K8s secrets from GSM in this and other projects"
       project_roles     = ["roles/secretmanager.secretAccessor"],
       cluster_namespace = "kubernetes-external-secrets"
+      additional_workload_identity_principals = [
+        "principalSet://iam.googleapis.com/${google_iam_workload_identity_pool_provider.eks_cluster.name}/attribute.sub/system:serviceaccount:default:default"
+      ]
     }
   }
 }
 
+
+resource "google_iam_workload_identity_pool" "eks_cluster" {
+  workload_identity_pool_id = "prow-eks"
+  display_name              = "EKS Prow Cluster"
+}
+
+resource "google_iam_workload_identity_pool_provider" "eks_cluster" {
+  workload_identity_pool_id          = google_iam_workload_identity_pool.eks_cluster.workload_identity_pool_id
+  workload_identity_pool_provider_id = "oidc"
+  attribute_mapping = {
+    "google.subject" = "assertion.sub"
+  }
+  oidc {
+    issuer_uri        = "https://oidc.eks.us-east-2.amazonaws.com/id/ASK MARKO"
```

Comment:
@xmudrii Can you get me OIDC endpoint?

`aws eks describe-cluster --name prow-build-cluster --region us-east-2 --query "cluster.identity.oidc"`